File: solutions/003/003.go
package main

import (
	"fmt"
)

func main() {
	fmt.Println(largestPrimeFactor(600851475143))
}

func largestPrimeFactor(number int64) int64 {
	var factor int64 = 1
	for i := int64(2); i*i <= number; i++ {
		if number == 1 {
			return factor
		} else if number%i != 0 {
			continue
		} else {
			factor = i
			for number%i == 0 {
				number /= i
			}
		}
	}
	return number
}

File: solutions/003/_003.java
public class _003 {
    public static void main(String[] args) {
        System.out.println(largestPrimeFactor(600851475143L));
    }

    public static long largestPrimeFactor(long number) {
        long factor = 1;
        for (int i = 3; i * i <= number; i += 2) {
            if (number == 1)
                return factor;
            else if (number % i != 0)
                continue;
            else {
                factor = i;
                while (number % i == 0)
                    number /= i;
            }
        }
        return number;
    }
}
